.github/workflows/github_push.yaml

```diff
@@ -1,68 +1,112 @@
-name: push
-on: [push]
+# Github worflow to test Agora on a variety of platforms
+#
+# The CI triggers on 'push' to upstream.
+#
+# Note:
+# - Try to use the native Github action syntax (${{ expression }}) when possible,
+#   as they are substituted with their value in the logs, unlike env variable.
+#   For example use `${{ github.workspace }}` over `${GITHUB_WORKSPACE}`
+name: CD pipeline
+
+on:
+  push:
+    branches:
+      - v*.x.x
+      - v*.*.x
+      # Use this branch name in your fork to test changes
+      - github-actions
 
 jobs:
-  test_mac:
-    name: MacOS
+  main:
+    name: Run
+    # The configuration matrix: We define all possible combinations here,
+    # then add excludes for things we don't want to test,
+    # and include to specify job-specific data.
     strategy:
+      # Disable `fail-fast` because we want the whole test suite to run even if one
+      # of the nigthly is broken
+      fail-fast: false
       matrix:
-        os: [macOS-15.04]
-        dc: [dmd-master, ldc-master]
+        os: [ ubuntu-18.04, macOS-10.15, windows-2019 ]
+        dc: [ dmd-master, ldc-master, dmd-latest ]
+        exclude:
+          # Not supported yet
+          - os: windows-2019
+          # We only build documentation on Linux
+          # Just use the latest release of DMD here, don't pin the version
+          - { os: macOS-10.15,  dc: dmd-latest }
+          - { os: windows-2019, dc: dmd-latest }
```

Comment:
If you only test the latest version here, you might want to use `Include:`

Reply:
`include` does not allow to add rows in a matrix

Reply:
https://help.github.com/en/actions/reference/workflow-syntax-for-github-actions#example-including-configurations-in-a-matrix-build
I think it's used like this.

Reply:
> You cannot use `include` to add new jobs to a build matrix.